helix-lsp/src/lib.rs

```diff
@@ -356,7 +356,13 @@ impl Registry {
                 let NewClientResult(client, incoming) = start_client(id, language_config, config)?;
                 self.incoming.push(UnboundedReceiverStream::new(incoming));
 
-                entry.insert((id, client.clone()));
+                let (_, old_client) = entry.insert((id, client.clone()));
+
+                tokio::spawn(async move {
```

Comment:
I wonder if we could add this to the Drop implementation instead on the server I guess?

Reply:
I don't have a way of replicating this unfortunately, but what I'm thinking is that if you impl Drop for the client, then you could just call shutdown_and_exist, then print something if needed. This approach may also solve https://github.com/helix-editor/helix/issues/3482

Reply:
I think the main issue there it's shutdown is async, I'll take a look at thsi when I have some time. This is also not a blocker, was just an improvement for better encapsulation IMO